@alwa-nordic please fix the build failure
```
zephyr/subsys/bluetooth/mesh/CMakeFiles/subsys__bluetooth__mesh.dir/settings.c.obj.d -o zephyr/subsys/bluetooth/mesh/CMakeFiles/subsys__bluetooth__mesh.dir/settings.c.obj -c /__w/zephyr/zephyr/subsys/bluetooth/mesh/settings.c
[154](https://github.com/zephyrproject-rtos/zephyr/runs/5217093609?check_suite_focus=true#step:6:154)
/__w/zephyr/zephyr/subsys/bluetooth/mesh/settings.c:18:10: fatal error: hci_core.h: No such file or directory
[155](https://github.com/zephyrproject-rtos/zephyr/runs/5217093609?check_suite_focus=true#step:6:155)
   18 | #include "hci_core.h"
[156](https://github.com/zephyrproject-rtos/zephyr/runs/5217093609?check_suite_focus=true#step:6:156)
      |          ^~~~~~~~~~~~
```

--- a/include/bluetooth/bluetooth.h
+++ b/include/bluetooth/bluetooth.h
@@ -130,10 +130,13 @@ typedef void (*bt_ready_cb_t)(int err);
  * Enable Bluetooth. Must be the called before any calls that
  * require communication with the local Bluetooth hardware.
  *
- * When @kconfig{CONFIG_BT_SETTINGS} has been enabled and the application is not
- * managing identities of the stack itself then the application must call
- * @ref settings_load() before the stack is fully enabled.
- * See @ref bt_id_create() for more information.
+ * When @kconfig{CONFIG_BT_SETTINGS} is enabled, the application must load the
+ * Bluetooth settings after this API call successfully completes before
+ * Bluetooth APIs can be used. Loading the settings before calling this function
+ * is insufficient. Bluetooth settings can be loaded with settings_load() or
+ * settings_load_subtree() with argument "bt". The latter selectively loads only
+ * Bluetooth settings and is recommended if settings_load() has been called
+ * earlier.
  *
  * @param cb Callback to notify completion or NULL to perform the
  * enabling synchronously.

--- a/subsys/bluetooth/host/settings.c
+++ b/subsys/bluetooth/host/settings.c
@@ -119,6 +119,16 @@ static int set(const char *name, size_t len_rd, settings_read_cb read_cb,
 	ssize_t len;
 	const char *next;
 
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_ENABLE)) {
+		/* The Bluetooth settings loader needs to communicate with the Bluetooth
+		 * controller to setup identities. This will not work before
+		 * bt_enable(). The doc on @ref bt_enable requires the "bt/" settings
+		 * tree to be loaded after @ref bt_enable is completed, so this handler
+		 * will be called again later.
+		 */
+		return 0;
+	}
+
 	if (!name) {
 		BT_ERR("Insufficient number of arguments");
 		return -ENOENT;
@@ -234,6 +244,16 @@ static int commit(void)
 	int err;
 
 	BT_DBG("");
+
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_ENABLE)) {
+		/* The Bluetooth settings loader needs to communicate with the Bluetooth
+		 * controller to setup identities. This will not work before
+		 * bt_enable(). The doc on @ref bt_enable requires the "bt/" settings
+		 * tree to be loaded after @ref bt_enable is completed, so this handler
+		 * will be called again later.
+		 */
+		return 0;
+	}
 
 #if defined(CONFIG_BT_DEVICE_NAME_DYNAMIC)
 	if (bt_dev.name[0] == '\0') {

--- a/subsys/bluetooth/mesh/settings.c
+++ b/subsys/bluetooth/mesh/settings.c
@@ -15,6 +15,7 @@
 #define LOG_MODULE_NAME bt_mesh_settings
 #include "common/log.h"
 
+#include "host/hci_core.h"
 #include "mesh.h"
 #include "subnet.h"
 #include "app_keys.h"
@@ -56,6 +57,16 @@ int bt_mesh_settings_set(settings_read_cb read_cb, void *cb_arg,
 
 static int mesh_commit(void)
 {
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_ENABLE)) {
+		/* The Bluetooth mesh settings loader calls bt_mesh_start() immediately
+		 * after loading the settings. This is not intended to work before
+		 * bt_enable(). The doc on @ref bt_enable requires the "bt/" settings
+		 * tree to be loaded after @ref bt_enable is completed, so this handler
+		 * will be called again later.
+		 */
+		return 0;
+	}
+
 	if (!bt_mesh_subnet_next(NULL)) {
 		/* Nothing to do since we're not yet provisioned */
 		return 0;

